src/js/fable-core/Set.fs

```diff
@@ -695,11 +696,19 @@ let createMutable (source: 'T seq) (comparer: IComparer<'T>) =
             upcast SetTree.mkIEnumerator tree
     }
 
-let distinct (xs: 'T seq) (comparer: IComparer<'T>): 'T seq =
+let distinct (xs: seq<'T>) ([<Inject>] comparer: IComparer<'T>) =
     createMutable xs comparer :> _ seq
 
-let distinctBy projection (xs: 'T seq) (comparer: IComparer<'T>): 'T seq =
-    createMutable (xs |> Seq.map projection) comparer :> _ seq
+let distinctBy (projection: 'T -> 'Key) (xs: seq<'T>) ([<Inject>] comparer: IComparer<'Key>) =
+    let hashSet = createMutable Seq.empty comparer
+    xs |> Seq.filter (projection >> hashSet.add)
```

Comment:
@ncave I found the issue: the `hashSet` reference was kept every time `Seq.filter` run away. So in the next iterations it discarded most of the times (the set already contained them). Pushing a fix right away.